src/Shipping/ShippingZone.php

```diff
@@ -78,6 +78,79 @@ public function get_shipping_methods_for_country( string $country_code ): array
 		return ! empty( $this->methods_countries[ $country_code ] ) ? array_values( $this->methods_countries[ $country_code ] ) : [];
 	}
 
+	/**
+	 * Gets the shipping rates for the given country.
+	 *
+	 * @param string $country_code
+	 *
+	 * @return array[] A multidimensional array of shipping rates for the given country. {
+	 *     Array of shipping method arguments.
+	 *
+	 *     @type string $country  The shipping method country.
+	 *     @type string $method   The shipping method ID.
+	 *     @type string $currency The currency which the shipping rate is in. Defaults to the store currency.
+	 *     @type float  $rate     The cost of the shipping method. Only if the method is flat-rate or local pickup.
+	 *     @type array  $options  Array of options for the shipping method (varies based on the method type). {
+	 *         Array of options for the shipping method.
+	 *
+	 *         @type array[] $shipping_class_rates An array containing the shipping class names and their rates. Only if the method is flat-rate.
+	 *         @type float   $free_shipping_threshold The minimum order amount required to use the shipping method. Only if the method is free shipping.
+	 *
+	 *     }
+	 * }
+	 */
+	public function get_shipping_rates_for_country( string $country_code ): array {
+		$methods       = $this->get_shipping_methods_for_country( $country_code );
+		$free_shipping = self::find_free_shipping_method( $methods );
+
+		$rates = [];
+		foreach ( $methods as $method ) {
+			// We process the free shipping method separately.
+			if ( ! $method['enabled'] || self::METHOD_FREE === $method['id'] ) {
+				continue;
+			}
+
+			// We can skip the pickup method because it's still not supported.
+			// Todo: Add support for the pickup method once it's available.
+			if ( self::METHOD_PICKUP === $method['id'] ) {
+				continue;
+			}
+
```

Comment:
The check for pickup method was added temporarily until we figure out if we can include the pickup method in this release. It seems like we can't so I removed it. The methods are already checked in `parse_method` function to see if we support them but just so that we have a more clear place, I've added a check in https://github.com/woocommerce/google-listings-and-ads/pull/1192/commits/ab8bf5552f485eda5318281d0836b4355b03568d.

Reply:
Thanks for the clarification!